python/api_server.py

```
@@ -0,0 +1,232 @@
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Payman transfer failed: {e}")
+
+    # ------------- Place order on Polymarket -------------
+    try:
+        from agents.polymarket.polymarket import Polymarket  # local module
+
+        poly = Polymarket()
+        market = poly.get_market(trade_request.market_id)
+        exec_resp = poly.execute_market_order([market], trade_request.amount)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Polymarket trade error: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)